lib/rules/max-nesting-depth/index.js

```diff
@@ -82,9 +82,12 @@ const rule = function(max, options) {
     }
 
     if (
-      optionsMatches(options, "ignore", "blockless-at-rules") &&
-      node.type === "atrule" &&
-      node.every(child => child.type !== "decl")
+      (optionsMatches(options, "ignore", "blockless-at-rules") &&
+        node.type === "atrule" &&
+        node.every(child => child.type !== "decl")) ||
+      (optionsMatches(options, "ignore", "blockless-pseudoclasses") &&
+        node.type === "rule" &&
+        node.some(child => child.selector && child.selector.startsWith("&:")))
```

Comment:
I think you're missing the `node.every(child => child.type !== "decl")` conditional to enforce the "and do not themselves have declaration blocks." bit of the description.

Reply:
Sorry I c&p the at-rules block, I have to remove that description line.